kong/plugins/ai-prompt-guard/handler.lua

```diff
@@ -1,112 +1,145 @@
-        buf:put(v.content)
+local execute do
+  local bad_format_error = "ai-prompt-guard only supports llm/v1/chat or llm/v1/completions prompts"
+
+  -- Checks the prompt for the given patterns.
+  -- _Note_: if a regex fails, it returns a 500, and exits the request.
+  -- @tparam table request The deserialized JSON body of the request
+  -- @tparam table conf The plugin configuration
+  -- @treturn[1] table The decorated request (same table, content updated)
+  -- @treturn[2] nil
+  -- @treturn[2] string The error message
+  function execute(request, conf)
+    local user_prompt
+
+    -- concat all 'user' prompts into one string, if conversation history must be checked
+    if type(request.messages) == "table" and not conf.allow_all_conversation_history then
+      local buf = buffer.new()
+
+      for _, v in ipairs(request.messages) do
+        if type(v.role) ~= "string" then
+          return nil, bad_format_error
+        end
+        if v.role == "user" then
+          if type(v.content) ~= "string" then
+            return nil, bad_format_error
+          end
+          buf:put(v.content)
+        end
       end
-    end
 
-    user_prompt = buf:get()
-
-  elseif request.messages then
-    -- just take the trailing 'user' prompt
-    for _, v in ipairs(request.messages) do
-      if v.role == "user" then
-        user_prompt = v.content
+      user_prompt = buf:get()
+
+    elseif type(request.messages) == "table" then
+      -- just take the trailing 'user' prompt
+      for _, v in ipairs(request.messages) do
+        if type(v.role) ~= "string" then
+          return nil, bad_format_error
+        end
+        if v.role == "user" then
+          if type(v.content) ~= "string" then
+            return nil, bad_format_error
+          end
+          user_prompt = v.content
+        end
       end
-    end
 
-  elseif request.prompt then
-    user_prompt = request.prompt
+    elseif type(request.prompt) == "string" then
+      user_prompt = request.prompt
 
-  else
-    return nil, "ai-prompt-guard only supports llm/v1/chat or llm/v1/completions prompts"
-  end
+    else
+      return nil, bad_format_error
+    end
+
+    if not user_prompt then
+      return nil, "no 'prompt' or 'messages' received"
+    end
 
-  if not user_prompt then
-    return nil, "no 'prompt' or 'messages' received"
-  end
 
-  -- check the prompt for explcit ban patterns
-  if conf.deny_patterns and #conf.deny_patterns > 0 then
-    for _, v in ipairs(conf.deny_patterns) do
+    -- check the prompt for explcit ban patterns
+    for _, v in ipairs(conf.deny_patterns or EMPTY) do
       -- check each denylist; if prompt matches it, deny immediately
       local m, _, err = ngx_re_find(user_prompt, v, "jo")
       if err then
-        return nil, "bad regex execution for: " .. v
+        -- regex failed, that's an error by the administrator
+        kong.log.err("bad regex pattern '", v ,"', failed to execute: ", err)
+        return kong.response.exit(500)
```

Comment:
Why do this here instead of executing `ngx_re_find("a", v, "jo")` in the schema validator for the `config.deny_patterns` field?